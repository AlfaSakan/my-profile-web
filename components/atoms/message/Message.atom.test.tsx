import renderer from 'react-test-renderer';
import Message from './Message.atom';

describe('message component test', () => {
  it('render correctly', () => {
    const tree = renderer
      .create(
        <Message
          message="message test"
          time={123456789}
          sender={{
            country_code: '+62',
            user_id: '1',
            name: 'admin',
            created_at: 123456789,
            image_url: '',
            phone_number: '8123456789',
            status: '',
            updated_at: 1213456789,
          }}
        />
      )
      .toJSON();

    expect(tree).toMatchSnapshot();
  });
});
