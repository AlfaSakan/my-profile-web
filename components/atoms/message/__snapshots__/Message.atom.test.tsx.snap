// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`message component test render correctly 1`] = `
<div
  className="bg-white h-fit min-w-fitCostume max-w-xl w-max px-2 py-1 relative rounded-md text-sm pr-16"
>
  <div
    className="absolute -top-2 -left-[0.9rem]"
  >
    <svg
      aria-hidden={true}
      className="MuiSvgIcon-root"
      focusable="false"
      style={
        Object {
          "fill": "white",
          "fontSize": 30,
        }
      }
      viewBox="0 0 24 24"
    >
      <path
        d="M14 7l-5 5 5 5V7z"
      />
    </svg>
  </div>
  <p
    className="text-xs font-semibold"
    style={
      Object {
        "color": undefined,
      }
    }
  >
    admin
  </p>
  <p>
    message test
  </p>
  <div
    className="absolute right-2 flex items-center bottom-0 gap-1"
  >
    <p
      className="text-xs"
    >
      17:17
    </p>
    <svg
      aria-hidden={true}
      className="MuiSvgIcon-root"
      focusable="false"
      style={
        Object {
          "fontSize": 14,
        }
      }
      viewBox="0 0 24 24"
    >
      <path
        d="M18 7l-1.41-1.41-6.34 6.34 1.41 1.41L18 7zm4.24-1.41L11.66 16.17 7.48 12l-1.41 1.41L11.66 19l12-12-1.42-1.41zM.41 13.41L6 19l1.41-1.41L1.83 12 .41 13.41z"
      />
    </svg>
  </div>
</div>
`;
